revisions-information.md

```diff
@@ -0,0 +1,87 @@
+# Hackage Metadata Revisions -- What They Are, How They Work
+
+## What are revisions?
+
+Package maintainers (as well as Hackage trustees) may provide metadata-revisions to existing package versions. These revisions can be thought of as "updating" the cabal file of a package with new or more current information. However, they do not _actually_ update the file. Tarballs retrieved from hackage are always as the authors uploaded them. Revision information is tracked explicitly in the Hackage index, and `cabal-install` will chose to prefer the latest revision of a cabal file to the original one when downloading or installing a package.
+
+## Why do we have revisions?
+
+Revisions are used to provide better metadata, typically in the case when a build-plan turns out to be incorrect, and we wish to tighten bounds to exclude incompatible versions of dependencies, or when a new version of a package is released and loosening dependencies will allow existing packages to work with it. Strategies for managing revisions and their costs are enumerated at: https://github.com/haskell-infra/hackage-trustees/blob/master/cookbook.md#best-practice-for-managing-meta-data
+
+Revisions can also be used when a package homepage or maintainer or the like changes, to keep information displayed current.
+
+## Why not just keep uploading new versions?
+
+First, uploading a whole bunch of code when only metadata changes leads to an unnecessary growth in versions. Second, often revisions need to be applied not only to the most recent version of a package, but prior versions as well. In particular, if a package at a given version has a bad install plan, then you do not want to let some tool continue to think this is a good plan, even if that package is not the latest version.
+
+## Where can I see revisions?
+
+Packages with revisions will tell you so on their cabal page. You can always access the revisions of a package by appending `/revisions` to its url, like so: http://hackage.haskell.org/package/lens-3.5.1/revisions/
+
+## How can I download a package without revisions applied?
+
+`cabal get --pristine`
+
+## How can I operate without taking into account revisions?
```

Comment:
What about "How can I ignore future revisions?"? You may still get some revisions with your current index-state.